[event-hubs] Final cleanup for preview (#5964)
Azure/azure-sdk-for-js: sdk/eventhub/event-hubs/README.md

### Publish events to an Event Hub

> In order to publish events, you'll need to create an `EventHubProducerClient`. Producers may be dedicated to a specific partition, or allow the Event Hubs service to decide which partition events should be published to. It is recommended to use automatic routing when the publishing of events needs to be highly available or when event data should be distributed evenly among the partitions. In the examples below, we will take advantage of automatic routing.

Reviewer: ```suggestion
In order to publish events, you'll need to create an `EventHubProducerClient`. You may publish events to a specific partition, or allow the Event Hubs service to decide which partition events should be published to. It is recommended to use automatic routing when the publishing of events needs to be highly available or when event data should be distributed evenly among the partitions. In the examples below, we will take advantage of automatic routing.
```

Reviewer: Fixed in https://github.com/Azure/azure-sdk-for-js/pull/5964/commits/fa55002b8df0ed60a4a3cb6fa5c258afe30a3d7c

#### Send a batch of events

Use the [createBatch](https://azuresdkdocs.blob.core.windows.net/$web/javascript/azure-event-hubs/5.0.0-preview.6/classes/eventhubproducer.html#createbatch) method to create

> an `EventDataBatch` object which can then be sent using the [sendBatch](https://azuresdkdocs.blob.core.windows.net/$web/javascript/azure-event-hubs/5.0.0-preview.6/classes/eventhubproducerclient.html#sendbatch) method.

Reviewer: Now that we are removing support for single event or array of events, this section needs to be as clear as possible. Right now it talks about create batch -> send -> add events. Lets change the order to create batch -> add events -> send batch

Reviewer: Fixed in https://github.com/Azure/azure-sdk-for-js/pull/5964/commits/fa55002b8df0ed60a4a3cb6fa5c258afe30a3d7c

Events may be added to the `EventDataBatch` using the [tryAdd](https://azuresdkdocs.blob.core.windows.net/$web/javascript/azure-event-hubs/5.0.0-preview.6/classes/eventdatabatch.html#tryadd)
method until the maximum batch size limit in bytes has been reached.

```javascript

> const { EventHubClient } = require("@azure/event-hubs");

Reviewer: ```suggestion
const { EventHubProducerClient } = require("@azure/event-hubs");
```

Reviewer: Fixed in https://github.com/Azure/azure-sdk-for-js/pull/5964/commits/fa55002b8df0ed60a4a3cb6fa5c258afe30a3d7c

async function main() {
  const producer = new EventHubProducerClient("connectionString", "eventHubName");

  const eventDataBatch = await producer.createBatch();
  
  let wasAdded = eventDataBatch.tryAdd({ body: "my-event-body" });
  wasAdded = eventDataBatch.tryAdd({ body: "my-event-body-2" });

> 

Reviewer: We are not checking the value of `wasAdded` before adding the second event...
I'd prefer if we change this to either
- Add a single event and then send the batch
- Or us an array of events, loop over it calling `tryAdd()` etc.

Thoughts?

Reviewer: Fixed in https://github.com/Azure/azure-sdk-for-js/pull/5964/commits/fa55002b8df0ed60a4a3cb6fa5c258afe30a3d7c

  await producer.sendBatch(eventDataBatch);

  await producer.close();
}

main();
```

The [Inspect an Event Hub](#inspect-an-event-hub) example shows how to get the list of partition ids should you wish to specify one for a producer.

The `EventHubProducerClient` constructor takes an optional parameter of type `EventHubClientOptions` which you can use to specify options like number of retries.

The `sendBatch` method takes an optional parameter of type `SendBatchOptions` which you can use to specify `abortSignal` to cancel current operation.

> All events that use the same partition key will be sent to the same partition.

Reviewer: Let's remove this line on partition key, it is not a concept we want to shout from the rooftops at the moment.
```suggestion
```

Reviewer: In https://github.com/Azure/azure-sdk-for-js/pull/5964/commits/fa55002b8df0ed60a4a3cb6fa5c258afe30a3d7c, I went ahead and added a section that talks about options at different levels which makes it more clear now.

**Note**: When working with Azure Stream Analytics, the body of the event being sent should be a JSON object as well.
For example: `body: { "message": "Hello World" }`

### Consume events from an Event Hub partition

> To consume events from a single Event Hub partition in a consumer group, create an [EventHubConsumerClient](https://azuresdkdocs.blob.core.windows.net/$web/javascript/azure-event-hubs/5.0.0-preview.6/classes/eventhubconsumerclient.html) for that partition and consumer group combination. 

Reviewer: > for that partition and consumer group combination

But we are not passing either to `EventHubConsumerClient`....

Reviewer: Instead of showcasing consuming events from a single partition, lets have only examples for the multi partition scenario. This translates to 2 sections
- Consume events from all partitions
- Consume events from all partitions with balanced load.

Reviewer: Fixed in https://github.com/Azure/azure-sdk-for-js/pull/5964/commits/2c47d67e3c7402e11860d09c0caea074547dc1be

const subscription = client.subscribe(
  EventHubClient.defaultConsumerGroupName,
  (receivedEvents, context) => { },
  partitionIds[0], {
    defaultEventPosition: EventPosition.latest()
  }
);

// events will be fired off as messages are received

> subscription.close();

Reviewer: Isnt this async? I yes, then we should add an await

Reviewer: On second thoughts... why do we need this code snippet here when we have the `Subscribe with an event handler` a few lines below?

```

> The [Inspect an Event Hub](#inspect-an-event-hub) example shows how to get the list of partition ids.

Reviewer: A few lines below there is a line `You can use this consumer in one of 3 ways to receive events:`

That was for the time we had pull, push and aync iterators.
Update this to reflect the 3 overloads to the `subscribe` method

Reviewer: Fixed in https://github.com/Azure/azure-sdk-for-js/pull/5964/commits/2c47d67e3c7402e11860d09c0caea074547dc1be
